src/hooks/useCalcLayout.ts

```diff
@@ -2,21 +2,33 @@ import React from 'react';
 
 import isEqual from 'lodash/isEqual';
 
-import type {Config} from '../shared';
+import type {Config, ConfigItem, ConfigLayout} from '../shared';
 import {RegisterManager} from '../utils';
 
 function onUpdatePropsConfig(config: Config, registerManager: RegisterManager) {
-    return config.layout.map((itemLayout, i) => {
-        const {type} = config.items[i];
-        return {
-            ...registerManager.getItem(type).defaultLayout,
-            ...itemLayout,
-        };
-    });
+    const configItems = [...config.items, ...(config.globalItems || [])];
+
+    return config.layout.reduce<ConfigLayout[]>((acc, itemLayout, i) => {
+        const item: ConfigItem | undefined = configItems[i];
+        const foundItem =
+            item && item.id === itemLayout.i
```

Comment:
`item && item.id === itemLayout.i` can be shorten as `item?.id === itemLayout.i`